replica/table.cc

```diff
@@ -2923,6 +2923,18 @@ write_memtable_to_sstable(memtable& mt, sstables::shared_sstable sst) {
     }
 }
 
+utils::small_vector<gate::holder, 1> table::hold_gate_for_compaction_groups(const dht::partition_range& pr) {
+    utils::small_vector<gate::holder, 1> async_gate_holders;
+    if (pr.is_singular()) {
+        async_gate_holders.push_back(compaction_group_for_token(pr.start()->value().token()).async_gate().hold());
+    } else {
+        for (auto& cg : compaction_groups_for_token_range(pr.transform(std::mem_fn(&dht::ring_position::token)))) {
```

Comment:
nit: we know the size we need in advance, so why not reserve async_gate_holders capacity as follows?
it would save reallocations and copies.
```c++
    auto compaction_groups = compaction_groups_for_token_range(pr.transform(std::mem_fn(&dht::ring_position::token)));
    async_gate_holders.reserve(compaction_groups.size());
    for (auto& cg : compaction_groups_for_token_range(pr.transform(std::mem_fn(&dht::ring_position::token)))) {
        async_gate_holders.push_back(cg->async_gate().hold());
    }
```

Reply:
Fixed in v4.